docs: vite 8.1 post (#22704)
vitejs/vite: docs/blog/announcing-vite8-1.md

---
title: Vite 8.1 is out!
author:
  name: The Vite Team

> date: 2026-06-18

Author: ```suggestion
date: 2026-06-22
```

sidebar: false
head:
  - - meta
    - property: og:type
      content: website
  - - meta
    - property: og:title
      content: Announcing Vite 8.1
  - - meta
    - property: og:image

>       content: https://vite.dev/og-image-announcing-vite8.webp

Author: TODO

  - - meta
    - property: og:url
      content: https://vite.dev/blog/announcing-vite8-1
  - - meta
    - property: og:description
      content: Vite 8 Release Announcement
  - - meta
    - name: twitter:card
      content: summary_large_image
---

# Vite 8.1 is out!

> _June 18, 2026_

Author: ```suggestion
_June 22, 2026_
```

![Vite 8 Announcement Cover Image](/og-image-announcing-vite8.webp) // TODO: update the cover image

> Vite 8 [was released](./announcing-vite8.md) in March with a single unified bundler powered by [Rolldown](https://rolldown.rs/), opening the door to further improvements. It is now seeing 27.7 million weekly downloads, around half of the total downloads of Vite 7. While we've focused on resolving upgrade regressions, we're excited to announce the release of Vite 8.1.

Author: ```suggestion
Vite 8 [was released](./announcing-vite8.md) in March with a single unified bundler powered by [Rolldown](https://rolldown.rs/), opening the door to further improvements. It is now seeing 34.8 million weekly downloads, around two third of the total downloads of Vite 7. While we've focused on resolving upgrade regressions, we're excited to announce the release of Vite 8.1.
```

::: details Why bundled dev mode?

Vite is known for its unbundled dev server approach, which is a main reason for Vite's speed and popularity when it was first introduced. This approach was initially an experiment to see just how far we could push the boundaries of development server performance without traditional bundling.

However, as projects scale in size and complexity, it has become clear that Vite's unbundled dev approach can degrade performance during development. Because each module is fetched separately, the browser must process a large number of requests, which increases startup and refresh overhead. This impact is especially noticeable in large applications and becomes more severe when developers are behind a network proxy, resulting in slower refresh times and a worse developer experience.

Bundled Dev Mode would allow serving bundled files not only in production but also during development, combining the best of both worlds:

- Fast startup times even for large applications
- Reduced network overhead on page refreshes
- Maintained efficient HMR on top of ESM output

:::

> Currently, it focuses on the browser side and the basic plugins and the main features. If you are using a third party plugin, it may not work with this mode. If you are using a minor feature, it may not work as well. We are working on expanding the support and preparing a document that clarifies the changes that may be needed on the plugin side. See [the design document](https://example.com) for more details about the roadmap.

Author: TODO

We are thinking of changing the default CSS preprocessor to Lightning CSS in the next major release. Please try it out by [`css.transformer: 'lightningcss'`](/config/shared-options#css-transformer) and share your feedback in [the discussion](https://github.com/vitejs/vite/discussions/13835).

### Case Insensitive Matching For `import.meta.glob`

`import.meta.glob` now supports `caseSensitive` option to match files case insensitively.

```ts
// matches ./dir/Module1.js
const modules = import.meta.glob('./dir/module*.js', {
  caseSensitive: false,
})
```

### Asset Discovery For Custom HTML Elements And Attributes

Previously, Vite would only discover assets for the elements and attributes that was pre-defined. Now, you can use the [`html.additionalAssetSources`](/config/shared-options#html-additionalassetsources) option to add more elements and attributes.

```html
<html-import src="./some/other/file.html"></html-import>
<img
  src="/layout-default.png"
  data-src-dark="/layout-dark.png"
  data-src-light="/layout-light.png"
/>
```

```ts [vite.config.js]
import { defineConfig } from 'vite'

export default defineConfig({
  html: {

>     additionalAssetSources: [

Reviewer: `[` is right?

Author: ah, it should be `{`. Thanks
